python/ray/air/checkpoint.py

```diff
@@ -230,6 +232,22 @@ def _metadata(self) -> _CheckpointMetadata:
             },
         )
 
+    def _copy_metadata_attrs_from(self, source: "Checkpoint") -> None:
+        """Copy in-place metadata attributes from ``source`` to self."""
+        for attr, value in source._metadata.checkpoint_state.items():
+            if attr in self._SERIALIZED_ATTRS:
+                setattr(self, attr, value)
+
+    @_metadata.setter
+    def _metadata(self, metadata: _CheckpointMetadata):
```

Comment:
When would you call this setter?

Reply:
This functionality is required for https://github.com/ray-project/ray/pull/29824